roofit/roostats/src/RooStatsUtils.cxx

```diff
@@ -330,3 +332,147 @@ namespace RooStats {
       os << ")\n";
    }
 }
+
+namespace {
+// freeing members from RooWorkspace:
+template <typename RooWorkspaceTag> struct RooWorkspaceHackResult {
+  typedef typename RooWorkspaceTag::type type;
+  static type ptr;
+};
+
+template <typename RooWorkspaceTag>
+typename RooWorkspaceHackResult<RooWorkspaceTag>::type
+    RooWorkspaceHackResult<RooWorkspaceTag>::ptr;
+
+template <typename RooWorkspaceTag, typename RooWorkspaceTag::type p>
+struct RooWorkspaceRob : RooWorkspaceHackResult<RooWorkspaceTag> {
+  struct RooWorkspaceFiller {
+    RooWorkspaceFiller() { RooWorkspaceHackResult<RooWorkspaceTag>::ptr = p; }
+  };
+  static RooWorkspaceFiller rooworkspacefiller_obj;
+};
+
+template <typename RooWorkspaceTag, typename RooWorkspaceTag::type p>
+typename RooWorkspaceRob<RooWorkspaceTag, p>::RooWorkspaceFiller
+    RooWorkspaceRob<RooWorkspaceTag, p>::rooworkspacefiller_obj;
+
+// now expose some members of RooWorkspace that we need to access
+struct RooWorkspace_snapshots {
+  typedef RooLinkedList(RooWorkspace::*type);
+};
+template struct RooWorkspaceRob<RooWorkspace_snapshots,
+                               &RooWorkspace::_snapshots>;
+} // namespace
```

Comment:
This horrible hack can be removed completely and instead be replaced by adding a function
```RooLinkedList RooWorkspace::getSnapshots(){ return this->_snapshots; }```
to RooWorkspace

Reply:
I agree ! the getSnapshots() has been added to rooworkspace.